driver/src/it/scala/org/mongodb/scala/DocumentationExampleSpec.scala

```diff
@@ -0,0 +1,516 @@
+    findObservable = collection.find(Filters.size("tags", 3))
+    //End Example 28
+
+    findObservable.execute().size should equal(1)
+  }
+
+  it should "query array of documents" in withCollection { collection =>
+
+    //Start Example 29
+    collection.insertMany(Seq(
+      Document("""{ item: "journal", instock: [ { warehouse: "A", qty: 5 }, { warehouse: "C", qty: 15 } ] }"""),
+      Document("""{ item: "notebook", instock: [ { warehouse: "C", qty: 5 } ] }"""),
+      Document("""{ item: "paper", instock: [ { warehouse: "A", qty: 60 }, { warehouse: "B", qty: 15 } ] }"""),
+      Document("""{ item: "planner", instock: [ { warehouse: "A", qty: 40 }, { warehouse: "B", qty: 5 } ] }"""),
+      Document("""{ item: "postcard", instock: [ { warehouse: "B", qty: 15 }, { warehouse: "C", qty: 35 } ] }""")
+    )).execute()
+    //End Example 29
+
+    collection.count().execute() should equal(5)
+
+    //Start Example 30
+    import org.mongodb.scala.model.Filters
+    var findObservable = collection.find(Filters.eq("instock", Document("warehouse" -> "A", "qty" -> 5)))
+    //End Example 30
+
+    findObservable.execute().size should equal(1)
+
+    //Start Example 31
+    findObservable = collection.find(Filters.eq("instock", Document("qty" -> 5, "warehouse" -> "A")))
+    //End Example 31
+
+    findObservable.execute().size should equal(0)
+
+    //Start Example 32
+    findObservable = collection.find(Filters.lte("instock.0.qty", 20))
+    //End Example 32
+
+    findObservable.execute().size should equal(3)
+
+    //Start Example 33
+    findObservable = collection.find(Filters.lte("instock.qty", 20))
+    //End Example 33
+
+    findObservable.execute().size should equal(5)
+
+    //Start Example 34
+    findObservable = collection.find(Filters.elemMatch("instock", Document("qty" -> 5, "warehouse" -> "A")))
+    //End Example 34
+
+    findObservable.execute().size should equal(1)
+
+    //Start Example 35
+    findObservable = collection.find(Filters.elemMatch("instock", Document("""{ qty: { $gt: 10, $lte: 20 } }""")))
+    //End Example 35
+
+    findObservable.execute().size should equal(3)
+
+    //Start Example 36
+    findObservable = collection.find(Filters.and(Filters.gt("instock.qty", 10), Filters.lte("instock.qty", 20)))
+    //End Example 36
+
+    findObservable.execute().size should equal(4)
+
+    //Start Example 37
+    findObservable = collection.find(Filters.and(Filters.eq("instock.qty", 5), Filters.eq("instock.warehouse", "A")))
+    //End Example 37
+
+    findObservable.execute().size should equal(2)
+  }
+
+  it should "query null and missing fields" in withCollection { collection =>
+
+    //Start Example 38
+    collection.insertMany(Seq(
+      Document("""{"_id": 1, "item": null}"""),
+      Document("""{"_id": 2}""")
+    )).execute()
+    //End Example 38
+
+    collection.count().execute() should equal(2)
+
+    //Start Example 39
+    var findObservable = collection.find(Document("item" -> None))
```

Comment:
In scala you don't use null. So here we are using the implicit BsonValue transformer to get BsonNull.